cluster-autoscaler/core/scaledown/actuation/update_latency_tracker_test.go

```diff
@@ -0,0 +1,112 @@
+package actuation
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/taints"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/test"
+)
+
+// mockClock is used to mock time.Now() when testing UpdateLatencyTracker
+// For the n th call to Now() it will return a timestamp after duration[n] to
+// the startTime if n < the length of durations. Otherwise, it will return current time.
+type mockClock struct {
+	startTime time.Time
+	durations []time.Duration
+	index     int
+	mutex     sync.Mutex
+}
+
+func NewMockClock(startTime time.Time, durations []time.Duration) mockClock {
+	return mockClock{
+		startTime: startTime,
+		durations: durations,
+		index:     0,
+	}
+}
+
+func (m *mockClock) Now() time.Time {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	var timeToSend time.Time
+	if m.index < len(m.durations) {
+		timeToSend = m.startTime.Add(m.durations[m.index])
+	} else {
+		timeToSend = time.Now()
+	}
+	m.index += 1
+	return timeToSend
+}
+
+func (m *mockClock) getIndex() int {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	return m.index
+}
+
+func TestUpdateLatencyCalculation(t *testing.T) {
+	toBeDeletedTaint := apiv1.Taint{Key: taints.ToBeDeletedTaint, Effect: apiv1.TaintEffectNoSchedule}
+	n1 := test.BuildTestNode("n1", 100, 100)
+	n1.Spec.Taints = append(n1.Spec.Taints, toBeDeletedTaint)
+	n2 := test.BuildTestNode("n2", 100, 100)
+	n2.Spec.Taints = append(n2.Spec.Taints, toBeDeletedTaint)
+	n3 := test.BuildTestNode("n3", 100, 100)
+
+	testCases := []struct {
+		description             string
+		startTime               time.Time
+		nodes                   []*apiv1.Node
+		durations               []time.Duration
+		wantLatency             time.Duration
+		wantResultChanNotClosed bool
```

Comment:
Why the negation? Can't we have `wantResultChanClosed` instead? This makes the test cases unnecessarily harder to read.

Reply:
Changed